includes/wp-admin/class-welcome-fields.php

```diff
@@ -0,0 +1,274 @@
+		<?php
+	}
+
+	/**
+	 * Render bookmarklet section.
+	 */
+	public static function render_bookmarklet_section() {
+		?>
+		<p>
+			<?php
+			$bookmarklet_js = get_reply_intent_js();
+
+			/* translators: %s is the domain of this site */
+			$reply_from_template = __( 'Reply from %s', 'activitypub' );
+			$button              = sprintf(
+				'<a href="%s" class="button">%s</a>',
+				esc_attr( $bookmarklet_js ), // Need to escape quotes for the bookmarklet.
+				sprintf( $reply_from_template, \wp_parse_url( \home_url(), PHP_URL_HOST ) )
+			);
+			/* translators: %s is where the button HTML will be rendered. */
+			$button_and_explanation_template = \__(
+				'%s Save this bookmarklet to reply to posts on other sites from your own blog! When visiting a post on another site, click the bookmarklet to start a reply.',
+				'activitypub'
+			);
+
+			printf( $button_and_explanation_template, $button ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+
+			printf( ' <a href="%s">%s</a>', esc_url( \admin_url( 'tools.php#activitypub' ) ), esc_html__( 'For additional information, please visit the Tools page.', 'activitypub' ) );
+			?>
+		</p>
+		<?php
+	}
+
+	/**
+	 * Render blog profile section.
+	 */
+	public static function render_blog_profile_section() {
+		$blog_user = new Blog();
+
+		?>
+		<p>
+			<?php \esc_html_e( 'People can follow your blog by using:', 'activitypub' ); ?>
+		</p>
+		<table class="form-table" role="presentation">
+			<tbody>
+				<tr>
+					<th scope="row"><?php \esc_html_e( 'Username', 'activitypub' ); ?></th>
+					<td>
+						<input type="text" class="large-text code" id="activitypub-blog-identifier" value="<?php echo \esc_attr( $blog_user->get_webfinger() ); ?>" readonly />
+					</td>
+				</tr>
+				<tr>
+					<th scope="row"><?php \esc_html_e( 'Profile URL', 'activitypub' ); ?></th>
+					<td>
+						<input type="text" class="large-text code" id="activitypub-blog-url" value="<?php echo \esc_attr( $blog_user->get_url() ); ?>" readonly />
+					</td>
+				</tr>
+			</tbody>
+		</table>
+		<p>
+			<?php \esc_html_e( 'This blog profile will federate all posts written on your blog, regardless of the author who posted it.', 'activitypub' ); ?>
+			<a href="<?php echo \esc_url_raw( \admin_url( '/options-general.php?page=activitypub&tab=blog-profile' ) ); ?>">
+				<?php \esc_html_e( 'Customize the blog profile', 'activitypub' ); ?>
+			</a>
+		</p>
+		<?php
+	}
+
+	/**
+	 * Render author profile section.
+	 */
+	public static function render_author_profile_section() {
+		$user = Actors::get_by_id( wp_get_current_user()->ID );
```

Comment:
```suggestion
		$user = Actors::get_by_id( wp_get_current_user_id() );
```
Not sure if it makes a big difference